spec.bs

```diff
@@ -0,0 +1,473 @@
+  The following step will be added to the <a href="https://fetch.spec.whatwg.org/#dom-request">new Request(input, init) constructor steps</a>, before step "Set this's [=Request/request=] to |request|":
+
+  1. If <var ignore=''>init</var>["{{RequestInit/browsingTopics}}"] <a for=map>exists</a>, then set |request|'s [=request/browsingTopic=] to it.
+
+  <span class=XXX>TODO: make the modification directly to the fetch spec.</span>
+
+  <h3 id="modification-to-http-network-or-cache-fetch-algorithm-header">Modification to HTTP-network-or-cache fetch algorithm</h3>
+  The following step will be added to the <a href="https://fetch.spec.whatwg.org/#concept-http-network-or-cache-fetch">HTTP-network-or-cache fetch</a> algorithm, before step "Modify |httpRequest|'s [=request/header list=] per HTTP. ...":
+
+  1. <a>Append or modify a request `<code>Sec-Browsing-Topics</code>` header</a> for |httpRequest|.
+
+  <span class=XXX>TODO: make the modification directly to the fetch spec.</span>
+
+  <h3 id="append-or-modify-a-request-sec-browsing-topics-header-header">Append or modify a request `Sec-Browsing-Topics` header</h3>
+  This specification defines a `<code><dfn http-header>Sec-Browsing-Topics</dfn></code>` [=header name=].
+
+  To <dfn>append or modify a request `<code>Sec-Browsing-Topics</code>` header</dfn>, given a [=request=] |request|, run these steps:
+    1. If |request|'s [=request/browsingTopics=] is not true, then return.
+    1. [=header list/Delete=] `<code>Sec-Browsing-Topics</code>` from |request|’s [=header list=].
+    1. Let |initiatorWindow| be |request|'s [=request/window=].
+    1. Let |requestOrigin| be |request|'s [=request/URL=]'s {{URL/origin}}.
+    1. If |requestOrigin| is not a [=potentially trustworthy origin=], then return.
+    1. If |initiatorWindow| is not an [=environment settings object=] whose [=environment settings object/global object=] is a {{Window}} object, then return.
+    1. If |initiatorWindow|'s <a data-cite="!HTML#concept-document-window">associated document</a> is not a [=secure context=], then return.
+    1. For each feature |f| in « "browsing-topic", "interest-cohort" »:
+        1. Run the <a href="https://www.w3.org/TR/permissions-policy-1/#algo-should-request-be-allowed-to-use-feature">Should request be allowed to use feature?</a> algorithm with <var ignore=''>feature</var> set to |f| and <var ignore=''>request</var> set to |request|. If the algorithm returns false, then return.
+
+        Note: the above algorithm should include the <a href="https://github.com/w3c/webappsec-permissions-policy/pull/499">pending update</a>, i.e. the |request| should be considered to contain the equivalent opt-in flags for both "browsing-topic" and the "interest-cohort" feature.
+    1. If the user preference setting disallows the access to topics from |topicsCallerContext|'s [=topics caller context/caller origin=] or |topicsCallerContext|'s [=topics caller context/top level context domain=], then return.
```

Comment:
This variable, _topicsCallerContext_, was never defined. Probably this check needs to be after the next few steps.